### DISCUSSION TITLE
Supporting Conda with Functions

### DISCUSSION THREAD

**Discussion by vrdmr** [Azure/azure-functions-python-worker#974] [Ideas]
Copying from: https://github.com/Azure/azure-functions-python-worker/issues/256

Making it a point of discussion and getting community feedback.

> This issue is a case for enabling conda/miniconda in the context of Python on Functions.
> 
> Investigative information
> A customer wanted to use fbprophet library which does not get installed through pip in both Windows and Mac. It depends on pystan library whose compilation options are best handed by conda.
> 
> Repro steps
> Create a new function, use HttpTrigger
> in init.py
> 
> import fbprophet
> 
> Now enable the virtual environment and do pip install fbprophet
> 
> Doesn't work.
> 
> Go out of the virtual environment and use conda install -c conda-forge fbprophet and it installs successfully. Conda knows the right kind of gcc flags needed to install pystan which is convenient for data scientists.
> 
> Either there must be some workaround to pip install fbprophet (even some compilation is fine) and a way to deploy such kinds of installations.
> 
> or support conda environments.
> 
> Currently, one customer is impacted, but fbprophet seems to be a library used for forecasting and is popular among certain customer domains.

**Comment by YaroslavKormushyn** [comment]
Hello!

I'm running into an issue trying to use Azure Functions in Python in order to query and display IFC files using [ifcopenshell](https://github.com/IfcOpenShell/IfcOpenShell) and [pythonocc-core](https://github.com/tpaviot/pythonocc-core).

Both of them aren't officially deployed to PyPi by their maintainers and the existing packages [here](https://pypi.org/project/pythonocc-core/) and [here](https://pypi.org/project/python-ifcopenshell/) are either out-of-date or simply aren't well put together (missing modules).

I see a great opportunity in supporting conda with these dependencies as their packages in the conda package index are maintained and prioritized by their authors. Otherwise, we're stuck with custom shell scripts to somehow download prebuilt packages for a particular OS (I'm developing on Win, the function is run on Linux) or have a static setup of packages in the SCM just for Linux deployment and do a manual setup of packages when developing on Windows.
